src/app/common/user.service.spec.js

```diff
@@ -0,0 +1,159 @@
+(function() {
```

Comment:
what Specifically ?
bro, if you test what happens if one variable is falsy, you should not test what happens if other variables are falsy in the same test, etc